cupyx/jit/_builtin_funcs.py

```diff
@@ -33,13 +44,6 @@ def call(self, env, *args, **kwargs):
         if step.ctype.dtype.kind not in 'iu':
             raise TypeError('range supports only for integer type.')
 
-        if isinstance(step, Constant):
-            step_is_positive = step.obj >= 0
-        elif step.ctype.dtype.kind == 'u':
-            step_is_positive = True
-        else:
-            step_is_positive = None
-
```

Comment:
Question: Why this block needs to be moved ahead?

Reply:
We should check whether `step` is `Constant` before `step = Data.init(step, env)`, since after this line it will be converted to `Data`.